catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/ProjectExporterTest.kt

```diff
@@ -0,0 +1,142 @@
+        Manifest.permission.WRITE_EXTERNAL_STORAGE,
+        Manifest.permission.READ_EXTERNAL_STORAGE
+    )
+
+    private lateinit var project: Project
+    private lateinit var contextMock: Context
+
+    @Before
+    fun setUp() {
+        project = Project(
+            ApplicationProvider.getApplicationContext(),
+            ProjectExporterTest::class.java.simpleName
+        )
+        ProjectManager.getInstance().currentProject = project
+        saveProjectSerial(project, ApplicationProvider.getApplicationContext())
+        val notificationManagerMock = Mockito.mock(NotificationManager::class.java)
+        contextMock = Mockito.mock(Context::class.java)
+        Mockito.`when`(contextMock.resources)
+            .thenReturn(ApplicationProvider.getApplicationContext<Context>().resources)
+        Mockito.`when`(contextMock.getSystemService(ArgumentMatchers.anyString()))
+            .thenReturn(notificationManagerMock)
+    }
+
+    @Test
+    fun exportProjectTest() {
+        createUndoCodeXmlFile()
+        val notificationManager = StatusBarNotificationManager(contextMock)
+        val notificationData = notificationManager.createSaveProjectToExternalMemoryNotification(
+            ApplicationProvider.getApplicationContext(),
+            project.name
+        )
+        ProjectExporter(
+            project.directory,
+            notificationData,
+            ApplicationProvider.getApplicationContext()
+        ).exportProjectToExternalStorage()
+        val externalProjectZip = File(
+            Constants.EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY,
+            project.directory.name + Constants.CATROBAT_EXTENSION
+        )
+        Assert.assertTrue(externalProjectZip.exists())
+        checkUndoCodeXmlFileIsDeleted(externalProjectZip)
+    }
+
+    private fun createUndoCodeXmlFile() {
+        val currentCodeFile = File(project.directory, Constants.CODE_XML_FILE_NAME)
+        val undoCodeFile = File(project.directory, Constants.UNDO_CODE_XML_FILE_NAME)
+        try {
+            StorageOperations.transferData(currentCodeFile, undoCodeFile)
+        } catch (exception: IOException) {
+            Log.e(TAG, "Copying project ${project.name} failed.", exception)
+        }
+    }
+
+    private fun checkUndoCodeXmlFileIsDeleted(externalProjectZip: File) {
+        val zipFileName =
+            externalProjectZip.absolutePath.replace(Constants.CATROBAT_EXTENSION, ".zip")
+        externalProjectZip.renameTo(File(zipFileName))
+        try {
+            val zipFile = ZipFile(zipFileName)
+            val zipEntries: Enumeration<*> = zipFile.entries()
+            var fileName: String
+            while (zipEntries.hasMoreElements()) {
+                fileName = (zipEntries.nextElement() as ZipEntry).name
+                Assert.assertFalse(fileName == Constants.UNDO_CODE_XML_FILE_NAME)
+            }
+        } catch (exception: IOException) {
+            Log.e(TAG, "Creating zip folder failed.", exception)
```

Comment:
Hmm, why should we catch an exception here? 
This could lead to a falsely passing test.